src/Service/SparkPostService.php

```diff
@@ -131,93 +203,220 @@ protected function prepareRecipients(Message $message): array
+
+            $recipient['address'] = $addressee;
+
+            // Format optional substitution_data-block
+            if ($message instanceof SparkPostMessage && $message->getVariables())
+            {
+                // Array of recipient-specific substitution variables indexed by email address
+                $substitutionVariables = $message->getVariables();
+
+                if (array_key_exists($addressee['email'], $substitutionVariables)) {
+                    $recipient['substitution_data'] = $substitutionVariables[$addressee['email']];
+                }
+            }
+
+            $recipients[] = $recipient;
         }
 
         return $recipients;
     }
 
     /**
-     * Prepare header string from message
-     *
-     * @param  Message $message
-     *
-     * @return string
+     * Prepare header structure from message
      */
-    protected function prepareHeaders(Message $message): string
+    protected function prepareHeaders(Message $message): array
     {
         $headers = clone $message->getHeaders();
-        $headers->removeHeader('Bcc');
 
-        return $headers->toString();
+        $removeTheseHeaders = [
+            'Bcc',
+            'Subject',
+            'From',
+            'To',
+            'Reply-To',
+            'Content-Type',
+            'Content-Transfer-Encoding',
+        ];
+
+        foreach ($removeTheseHeaders as $headerName) {
+            $headers->removeHeader($headerName);
+        }
+
+        return $headers->toArray();
+    }
+
+    /**
+     * Prepare the 'metadata' structure for the SparkPost Transmission call
+     */
+    protected function prepareMetadata(Message $message): array
+    {
+        $metadata = [];
+
+        if ($message->getSubject()) {
+            $metadata['subject'] = $message->getSubject();
+        }
+
+        if ($message->getSender()) {
+            $sender = $message->getSender();
+
+            if($sender instanceof AddressInterface) {
+                $metadata['from'] = [];
+                $metadata['from']['email'] = $sender->getEmail();
+                $metadata['from']['name'] = $sender->getName() ?: $sender->getEmail();
+            }
+        }
+
+        if ($message->getReplyTo()) {
+            $replyToList = $message->getReplyTo();
+            $replyToList->rewind();
+            $replyToAddress = $replyToList->current();
+
+            if ($replyToAddress instanceof AddressInterface) {
+                $metadata['reply_to'] = $replyToAddress->getName() ? $replyToAddress->toString() : $replyToAddress->getEmail();
+            }
+        }
+
+        return $metadata;
     }
 
     /**
      * Prepare body string from message
-     *
-     * @param  Message $message
-     *
-     * @return string
      */
     protected function prepareBody(Message $message): string
     {
         return $message->getBodyText();
     }
 
+    public function registerSendingDomain(string $domain): bool
```

Comment:
@nickvanderveeken I think it makes more sense to simply pass the DKIM parameter as (optional) arguments to this method. I think in general the DKIM settings belong to this method call; and there is no need for a _default_ DKIM config. 

I could be missing something here; so please let me know if that is the case!

Reply:
Yeah, it makes more sense.